engine/src/main/java/org/terasology/entitySystem/event/AbstractBeforeAfterEvent.java

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.entitySystem.event;
+
+/**
+ * Immutable event to notify a change in a certain value.
+ * NOTE: This is a generic event and a more specific Event extending this event should be fired and reacted to.
+ */
+public class AbstractBeforeAfterEvent implements Event {
+    private final float oldValue;
+    private final float newValue;
```

Comment:
If the class works with a generic `<T>` these values become of type `T`.
```suggestion
    protected final T oldValue;
    protected final T newValue;
```

If subclasses should be able to access `oldValue` and `newValue` those should be protected instead of private.

Reply:
Ah yes. I should've at least gotten that right 😔